src/test/ui/impl-trait/where-allowed.stderr

```diff
@@ -110,184 +110,207 @@ LL | fn in_impl_Fn_parameter_in_return() -> &'static impl Fn(impl Debug) { panic
+   |           ^^^^^^^^^^
+
+error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
+  --> $DIR/where-allowed.rs:240:40
+   |
+LL | fn in_method_generic_param_default<T = impl Debug>(_: T) {}
+   |                                        ^^^^^^^^^^
+
+error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
+  --> $DIR/where-allowed.rs:246:29
    |
 LL |     let _in_local_variable: impl Fn() = || {};
    |                             ^^^^^^^^^
    |
    = help: add `#![feature(impl_trait_in_bindings)]` to the crate attributes to enable
 
 error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
-  --> $DIR/where-allowed.rs:224:46
+  --> $DIR/where-allowed.rs:248:46
    |
 LL |     let _in_return_in_local_variable = || -> impl Fn() { || {} };
    |                                              ^^^^^^^^^
 
-error[E0720]: cannot resolve opaque type
-  --> $DIR/where-allowed.rs:56:49
-   |
-LL | fn in_impl_Fn_parameter_in_return() -> &'static impl Fn(impl Debug) { panic!() }
-   |                                                 ^^^^^^^^^^^^^^^^^^^   -------- this returned value is of `!` type
-   |                                                 |
-   |                                                 cannot resolve opaque type
-   |
-   = help: this error will resolve once the item's body returns a concrete type
-
-error[E0720]: cannot resolve opaque type
-  --> $DIR/where-allowed.rs:62:46
+error: defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
+  --> $DIR/where-allowed.rs:234:7
    |
-LL | fn in_impl_Fn_return_in_return() -> &'static impl Fn() -> impl Debug { panic!() }
-   |                                              ^^^^^^^^^^^^^^^^^^^^^^^   -------- this returned value is of `!` type
-   |                                              |
-   |                                              cannot resolve opaque type
+LL | impl <T = impl Debug> T {}
+   |       ^
    |
-   = help: this error will resolve once the item's body returns a concrete type
+   = note: `#[deny(invalid_type_param_default)]` on by default
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #36887 <https://github.com/rust-lang/rust/issues/36887>
 
-error: could not find defining uses
-  --> $DIR/where-allowed.rs:121:16
+error: defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
+  --> $DIR/where-allowed.rs:240:36
    |
-LL |     type Out = impl Debug;
-   |                ^^^^^^^^^^
-
-error: could not find defining uses
-  --> $DIR/where-allowed.rs:157:23
+LL | fn in_method_generic_param_default<T = impl Debug>(_: T) {}
+   |                                    ^
    |
-LL | type InTypeAlias<R> = impl Debug;
-   |                       ^^^^^^^^^^
```

Comment:
I think these errors disappeared because hitting the `invalid_type_param_default` future compat lint is making the compiler abort earlier.